R/verb-copy-to.R

```diff
@@ -87,6 +87,64 @@ copy_to.src_sql <- function(dest, df, name = deparse(substitute(df)),
   invisible(out)
 }
 
+#' @export
+db_values <- function(con, df, name = deparse(substitute(df))) {
+  if (ncol(df) == 0) {
+    abort("`df` needs at least one column.")
+  }
+
+  if (!inherits(df, "data.frame")) {
+    abort("`df` needs to be a data.frame.")
+  }
+
+  tbl(
+    src = con,
+    from = sql_values(con, df),
+    vars = colnames(df)
+  )
+}
+
+sql_values <- function(con, df) {
+  UseMethod("sql_values")
+}
+
+#' @export
+sql_values.DBIConnection <- function(con, df) {
+  sql_values_clause(con, df)
+}
+
+sql_values_clause <- function(con, df, row = FALSE) {
+  sim_data <- rep_named(colnames(df), list(NULL))
+  cols_clause <- escape(sim_data, con = con, parens = FALSE)
+  null_row_clause <- build_sql("SELECT ", cols_clause, " WHERE false", con = con)
+
+  if (nrow(df) == 0) {
+    return(null_row_clause)
+  }
+
+  values <- purrr::map(df, escape, con = con, collapse = NULL, parens = FALSE)
+  rows <- purrr::pmap(
```

Comment:
Can we use a variant of `rlang::exec(paste, values_with_comma)` here? The new `values_with_comma` would contain the first data item followed by a comma followed by the second data item ... . This would make use of vectorization.

Reply:
I used the idea in the new approach, thanks!